### PR TITLE
C4R-74 View All apps (as Cards)

### PR DESCRIPTION
**Pull request by dragos1195**
C4R-74 

Added View Applications as Cards

All apps for the moment.

### CODE REVIEW

**Comment by dragos1195** [comment]
WIP

File: frontend/src/pages/requests/components/Request.tsx [hunk 1]
```diff
@@ -34,21 +34,20 @@ const Request = () => {
   const {
     data: request,
     error,
-    refetch,
     isLoading: dataLoading,
-  } = useRequest(params.id ? params?.id : '');
+  } = useOrganizationRequest(params.id ? params?.id : '');
```
Comment (radulescuandrew) [inline frontend/src/pages/requests/components/Request.tsx:38]: Nu stiu daca "params" poate sa fie undefined, dar merita sa pui optional chaining si la "params.id" sau sa folosesti `id` destructurat pe linia 22.

File: frontend/src/pages/requests/interfaces/OngApplication.interface.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  ACTIVE = 'active',
+  PENDING = 'pending',
+  RESTRICTED = 'restricted',
+}
+
+export interface OngApplication {
+  organizationId: number;
+  organization: IOrganization;
+
+  applicaiontId: number;
```
Comment (radulescuandrew) [inline frontend/src/pages/requests/interfaces/OngApplication.interface.ts:14]: application*

File: frontend/src/services/application/Application.queries.ts [hunk 3]
```diff
@@ -41,7 +50,25 @@ export const useApplicationsQuery = (
   );
 };
 
-export const useApplication = (applicationId: string) => {
+export const useOngApplicationsQuery = () => {
```
Comment (radulescuandrew) [inline frontend/src/services/application/Application.queries.ts:53]: Nu inteleg diferenta dintre `useOngApplicationsQuery` si `useMyOngApplicationsQuery` dpdv business logic. 

Ar trebui sa punem comments prin multe locuri in app.

File: frontend/src/services/application/Application.service.ts [hunk 4]
```diff
@@ -34,6 +39,24 @@ export const getApplications = async (
   return API.get(requestUrl).then((res) => res.data);
 };
 
+export const getOngApplications = async (): Promise<PaginatedEntity<ApplicationWithOngStatus>> => {
```
Comment (radulescuandrew) [inline frontend/src/services/application/Application.service.ts:42]: Nici /organization/application si /organization/application/profile nu inteleg

File: frontend/src/services/request/Request.queries.ts [hunk 5]
```diff
@@ -57,23 +75,69 @@ export const useRequest = (requestId: string) => {
...
+  limit: number,
+  page: number,
+  orderBy: string,
+  orderDirection: OrderDirection,
+  search?: string,
+  interval?: Date[],
+) => {
+  const { setApplicationRequests } = useStore();
+  return useQuery(
+    ['applications/requests', limit, page, orderBy, orderDirection, search, interval],
```
Comment (radulescuandrew) [inline frontend/src/services/request/Request.queries.ts:119]: Stringurile astea `applications/requests` apeleaza API-ul nu?

**Review by radulescuandrew — CHANGES_REQUESTED** [review]

File: frontend/src/pages/application/Application.tsx [hunk 6]
```diff
@@ -41,12 +44,17 @@ const Application = () => {
     return <Loading />;
   }
 
+  const naivgateBack = () => {
+    // TEMPORARY - needs changes
+    navigate(-2);
```
[the remark this answers is not in the document]
Comment (dragos1195) [reply frontend/src/pages/application/Application.tsx:49]: https://wearetribus.atlassian.net/browse/C4R-616

File: frontend/src/services/request/Request.queries.ts (hunk 5 above)
Comment (dragos1195) [reply frontend/src/services/request/Request.queries.ts:119]: Nu. Sunt query keys. 
React query administreaza cache-ul pe baza de query keys. 
Functia apelata este mai jos.

https://react-query-v2.tanstack.com/guides/query-keys#_top

File: frontend/src/pages/application/Application.tsx [hunk 7]
```diff
@@ -55,32 +63,34 @@ const Application = () => {
       }}
       deleteButton={{
```
[the remark this answers is not in the document]
Comment (dragos1195) [reply frontend/src/pages/application/Application.tsx]: Removed.

File: frontend/src/pages/apps-store/AllApplications.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+import { UserRole } from '../users/enums/UserRole.enum';
+import { APPLICATION_STORE_TABS } from './constants/ApplicationStoreTabs.constant';
+
+const AllApplications = () => {
+  const navigate = useNavigate();
+  const [selectedTab, setSelectedTab] = useState(0);
+  const { role } = useAuthContext();
+
+  useEffect(() => {
+    const found: IPageTab | undefined = APPLICATION_STORE_TABS.find(
```
[the remark this answers is not in the document]
Comment (dragos1195) [reply frontend/src/pages/apps-store/AllApplications.tsx:16]: Added comment.

File: frontend/src/pages/apps-store/components/ApplicationListCards.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import { Loading } from '../../../components/loading/Loading';
+import {
+  useMyOngApplicationsQuery,
+  useOngApplicationsQuery,
+} from '../../../services/application/Application.queries';
+import { ApplicationWithOngStatus } from '../../../services/application/interfaces/Application.interface';
+import { useOngApplications } from '../../../store/selectors';
+import ApplicationCard from '../../my-apps/components/ApplicationCard';
+
+const ApplicationListCards = ({ ong }: { ong?: boolean }) => {
```
[the remark this answers is not in the document]
Comment (dragos1195) [reply frontend/src/pages/apps-store/components/ApplicationListCards.tsx]: Changed to isOngView. 

Pentru MyApps si AllApps as Admin sunt tot Cards. 
Am pus un parametru sa diferentiez ce call fac.

File: frontend/src/services/application/Application.queries.ts (hunk 3 above)
Comment (dragos1195) [reply frontend/src/services/application/Application.queries.ts:51]: Am adaugat comentarii la definitie

File: frontend/src/services/application/Application.service.ts (hunk 4 above)
Comment (dragos1195) [reply frontend/src/services/application/Application.service.ts:43]: Adaugat comentarii

**Pull request merged**